kit/Kit.cpp

```diff
@@ -360,16 +360,12 @@ namespace
             if (!strcmp(dot, ".so"))
             {
                 // NSS is problematic ...
-                if (strstr(path, "libnspr4") ||
-                    strstr(path, "libplds4") ||
-                    strstr(path, "libplc4") ||
-                    strstr(path, "libnss3") ||
-                    strstr(path, "libnssckbi") ||
-                    strstr(path, "libnsutil3") ||
-                    strstr(path, "libssl3") ||
-                    strstr(path, "libsoftokn3") ||
-                    strstr(path, "libsqlite3") ||
-                    strstr(path, "libfreeblpriv3"))
+                if (strstr(path, "libnspr4") || strstr(path, "libplds4") ||
+                    strstr(path, "libplc4") || strstr(path, "libnss3") ||
+                    strstr(path, "libnssckbi") || strstr(path, "libnsutil3") ||
+                    strstr(path, "libssl3") || strstr(path, "libsoftokn3") ||
+                    strstr(path, "libsqlite3") || strstr(path, "libfreeblpriv3") ||
+                    strstr(path, "libucphier1"))
```

Comment:
This bit strikes me as suspicious, because everything else listed here is belongs to nss (or nss links to it) and we don't list any other of LibreOffice's own shared libraries here.

IIUC we "preload" a bunch of libraries and I suspect that this has a relation so the problem you are seeing. Looking at core I see a twisty set of things in lo_initialize in desktop/source/lib/init.cxx but from there then in "cppuhelper/source/servicemanager.cxx" I see:

/// Allow external callers & testers to disable certain components
const char *pDisable = getenv("UNODISABLELIBRARY");

and I see in kit/Kit.cpp we have:

// Disable problematic components that may be present from a
// desktop or developer's install if env. var not set.
::setenv("UNODISABLELIBRARY",
         ... ucpgio1 ucphier1 ucpimage ...
             0 /* no overwrite */);

Which I think is intended to disable libraries that might happen to be in a developers local build of core, that don't exist in the real/release configuration core build. So... I wonder if removing "ucphier1" from that other list instead of adding libucphier1 to this list also fixes the problem?

If that also works, I expect that is the preferred solution. (And, assuming it works, worth a separate pr which we can run by Michael to see if ucphier1 is just accidentally added in there or if there is a reason I'm unaware of why that is explicitly disabled in that UNODISABLELIBRARY list)

Reply:
Yes it works. I have created https://github.com/CollaboraOnline/online/pull/10618